pallets/parachain-staking/src/lib.rs

```diff
@@ -968,15 +968,15 @@ pub mod pallet {
 						Delegator state also has a record. qed.",
 				);
 				if let Some(remaining) = delegator.rm_delegation(&candidate) {
+					Self::delegation_remove_request_with_state(
+						&candidate,
+						&bond.owner,
+						&mut delegator,
+					);
+
 					if remaining.is_zero() {
 						<DelegatorState<T>>::remove(&bond.owner);
```

Comment:
in this branch, we should be removing all requests for the delegator because the delegator is leaving. It isn't done here -- is it not a problem @nbaztec

Reply:
I think it is fine because it implies this is the last delegation so therefore it is the last pending delegation request. A comment would be nice though tbh, for posterity.

Reply:
Yes, this assumes no inconsistency. Can add a comment.